website/pages/docs/connect/observability/ui-visualization.mdx

```diff
@@ -0,0 +1,501 @@
+`http://prometheus-server` (using Kubernetes' DNS resolution).
+
+A full configuration to enable Prometheus is given below.
+
+```hcl
+ui_config {
+  enabled = true
+  metrics_provider = "prometheus"
+  metrics_proxy {
+    base_url = "http://prometheus-server"
+  }
+}
+```
+
+## Configuring Dashboard URLs
+
+Since Consul's visualization is intended as an overview of your mesh and not a
+comprehensive monitoring tool, you can configure a service dashboard URL
+template which allows users to click directly through to the relevant
+service-specific dashboard in an external tool like Grafana or a hosted
+provider.
+
+To configure this, you must provide a URL template in the [agent configuration
+file](/docs/agent/options#ui_config_dashboard_url_templates) for all agents that
+have the UI enabled. The template is essentially the URL to the external
+dashboard, but can have placeholder values which will be replaced with the
+service name, namespace and datacenter where appropriate to allow deep-linking
+to the relevant information.
+
+An example with Grafana is shown below.
+
+```hcl
+ui_config {
+  enabled = true
+  dashboard_url_templates {
+    service = "https://grafana.example.com/d/lDlaj-NGz/ ⏎
+      service-overview?orgId=1&var-service={{Service.Name}}& ⏎
+      var-namespace={{Service.Namespace}}&var-dc={{Datacenter}}"
+  }
+}
+```
+
+TODO: show screenshot of the configured link in the topology view.
+
+
+### Metrics Proxy
+
+In many cases the metrics backend may not be exposed outside of the datacenter
+to user's browser or may be on a different domain and so subject to CORS
+restrictions. To make it simpler to serve the metrics to the UI in these cases,
+the Consul agent can proxy requests for metrics from the UI to the backend.
+
+**This is intended to enable simple setup in test and demo environments. Careful
+consideration should be given towards using this in production.**
+
+The simplest configuration is described in [Configuring the UI for
+metrics](#configuring-the-ui-for-metrics).
+
+#### Metrics Proxy Security
+
+~> **Security Note**: Exposing a backend metrics service to potentially
+ un-authenticated network traffic via the proxy should be _carefully_ considered
+ in production.
+
+The proxy endpoint is an internal one intended only for UI use however by
+enabling it anyone with network access to the agent's API port is potentially
+able to access metrics from the backend.
+
+**If ACLs are not enabled, full access to metrics will be exposed to
+un-authenticated workloads on the network**.
+
+With ACLs enabled, the proxy endpoint requires a valid token with read access to
+all nodes and services (across all namespaces in Enterprise). It's typical for
+most authenticated users to have this level of access in Consul as it's required
+for viewing the catalog or discovering services. For example if you use a
+[Single Sign-On
+integration](http://localhost:3000/docs/security/acl/auth-methods/oidc) (Consul
```

Comment:
Includes full link to `localhost` for Single Sign-On, I think this should just be `/docs/security/acl/auth-methods/oidc`?